syntax = "proto3";

package pythonservice; 

service PythonRunner {
  rpc ExecuteScript(ScriptRequest) returns (ScriptResponse); 
  rpc ExecuteScriptStream(ScriptRequest) returns (stream ScriptResponse) ;
}

message ScriptRequest {
        string script_name = 1 ; 
        repeated string args = 2 ;
        map<string , string> env_vars = 3;
}


message ScriptResponse{
        int32 exit_code = 1;
        string stdout =2 ; 
        string stderr = 3;
        bool success = 4;
}



